### ISSUE TITLE
Terminal shell integration not available on window reload

### ISSUE THREAD

**Issue by njiang747** [microsoft/vscode#236927]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.94.0
- OS Version: macOS Sonoma 14.4, Ubuntu 20.04

Steps to Reproduce:

1. Reload window with active terminals
2. Check `terminal.shellIntegration` for active terminals (will be `undefined`)

Seems like `terminal.shellIntegration` is not getting set even though the shell integration is supported. If a command is executed in the terminal (even if it's just `^C`), the shell integration will be set. Ideally the terminal's shell integration would be set on reload (or unset immediately at startup then set and triggering `vscode.window.onDidChangeTerminalShellIntegration` at some point after).

It seems like this was acknowledged before given the code https://github.com/microsoft/vscode/blob/main/src/vs/workbench/api/common/extHostTerminalShellIntegration.ts#L108-L112 which explicitly triggers `$shellIntegrationChange` on `$shellExecutionStart` if the shell integration hasn't been set yet (and the accompanying comment).

**Comment by Tyriar** [comment]
If I understand right this is by design, `shellIntegration` will never be set when the terminal starts up, even if it's restored and had it before. You must always listen to the event

**Comment by njiang747** [comment]
@Tyriar Agreed that on initial startup shell integration is expected to be unset. I guess the thing that seems unexpected to me is that the `onDidChangeTerminalShellIntegration` isn't fired until a command is actually executed in the terminal.

**Comment by Tyriar** [comment]
@njiang747 understood, we actually fixed this yesterday in https://github.com/microsoft/vscode/pull/237524 👍

**Comment by njiang747** [comment]
@Tyriar oh awesome, thanks!
